cohort-engine/src/main/java/com/ibm/cohort/engine/BooleanEvaluationCallback.java

```diff
@@ -0,0 +1,49 @@
+/*
+ *
+ *  * (C) Copyright IBM Corp. 2021
+ *  *
+ *  * SPDX-License-Identifier: Apache-2.0
+ *
+ */
+
+package com.ibm.cohort.engine;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.slf4j.Logger;
+
+import org.slf4j.LoggerFactory;
+
+public class BooleanEvaluationCallback implements EvaluationResultCallback {
+
+
+	public List<String> getPassingPatients(){
+		return passingPatients;
+	}
+
+	private List<String> passingPatients = new ArrayList<>();
+	private static final Logger logger = LoggerFactory.getLogger(BooleanEvaluationCallback.class.getName());
+
+	@Override
+	public void onContextBegin(String contextId) {
+		logger.info("Context: " + contextId);
+	}
+
+	@Override
+	public void onContextComplete(String contextId) {
+		logger.info("---");
+	}
+
+	@Override
+	public void onEvaluationComplete(String contextId, String expression, Object result) {
+		if( result != null ) {
+			if (result instanceof Boolean) {
+				if((Boolean) result){
+					passingPatients.add(contextId);
+				}
+				logger.info(String.format("Expression: \"%s\", Result: %s", expression, result));
+			}
```

Comment:
Would it make sense to throw an error if a boolean is not returned here?
A user accidentally submitting a CQL define that doesn't return a boolean value seems very possible, and could cause confusion when the result mimics if a patient simply wasn't in the measure (aside from the missing log statement).
I'm leaning towards failing hard and fast here, but I'd like to hear any other opinions on this as well.

I'm not sure if we want to also do something similar for `null`.
If we want `null` to be a legit value, then are we OK with it being functionally equivalent to `false` (aside from the missing log statement)?
My gut says to fail hard and fast here as well until we identify a real usecase where `null` is a sensible return value.

Reply:
I think I had that at some point, but cut it out when shuffling things around. I agree that it's a good idea to fail loudly, and have added that